webapp/components/_new/generic/UserProfileHeader/UserProfileHeader.vue

```diff
@@ -0,0 +1,244 @@
+        return {
+          // Still setting some height in case the user uploads a super super wide picture.
+          // In that case the height would, in theory, be able to go to 0 thus changing a profile
+          // picture won't be possible anymore.
+          '--min-height': 20 + 'px',
+          '--max-height': 250 + 'px',
+        }
+      } else {
+        // If no picture is present, the height is fixed to 250 so user can click the upload zone.
+        return {
+          '--height': 250 + 'px',
+        }
+      }
+    },
+  },
+  watch: {
+    error() {
+      const that = this
+      setTimeout(function () {
+        that.error = false
+      }, 2000)
+    },
+  },
+  methods: {
+    template() {
+      return `<div class="dz-preview dz-file-preview">
+                <div class="dz-image">
+                  <div data-dz-thumbnail-bg></div>
+                </div>
+              </div>
+      `
+    },
+    vddrop(file) {
+      const profileHeaderUpload = file[0]
+      this.$apollo
+        .mutate({
+          mutation: updateUserMutation(),
+          variables: {
+            profileHeader: {
+              upload: profileHeaderUpload,
+            },
+            id: this.user.id,
+          },
+        })
+        .then(() => {
+          this.$toast.success(this.$t('user.profileHeader.submitted'))
+        })
+        .catch((error) => this.$toast.error(error.message))
+    },
+    verror(file, message) {
+      if (file.status === 'error') {
+        this.error = true
+        this.$toast.error(file.status, message)
+      }
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+.profile-header {
+  min-height: var(--min-height);
+  max-height: var(--max-height);
+  height: var(--height);
+  background-color: DarkGrey; /* Fallback color */
+}
+
+.profile-header-image {
+  width: 100%;
+}
+
+#profileHeaderDropzone {
+  height: 100%;
+}
+
+.dz-message {
+  height: 100%;
+}
+
+#profileHeaderDropzone .dz-preview {
+  transition: all 0.2s ease-out;
+  width: 160px;
+  display: flex;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image {
+  width: 100%;
+  height: 100%;
+  object-fit: contain;
+  overflow: hidden;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image > div {
+  width: inherit;
+  height: inherit;
+  border-radius: 50%;
+  background-size: cover;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image > img {
+  width: 100%;
+}
+
+.profileHeader-attachments-upload-area {
+  height: 100%;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+}
+
+.profileHeader-attachments-upload-button {
+  pointer-events: none;
+}
+
+.profileHeader-drag-marker {
+  position: relative;
+  width: 122px;
+  height: 122px;
+  border-radius: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  color: hsl(0, 0%, 25%);
+  transition: all 0.2s ease-out;
+  font-size: 60px;
+
+  background-color: rgba(255, 255, 255, 0.7);
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-06-18T09:51:17Z" title="Thursday, June 18th 2020, 11:51:17 am +02:00">Jun 18, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,224 @@
+<template>
+  <div class="profile-header">
+    <vue-dropzone
+      v-if="user && this.editable"
+      id="profileHeaderDropzone"
+      :key="profileHeaderUrl"
+      ref="el"
+      :use-custom-slot="true"
+      :options="dropzoneOptions"
+      @vdropzone-error="verror"
+    >
+      <div class="dz-message" @mouseover="hover = true" @mouseleave="hover = false">
+        <img
+          v-if="user && user.profileHeader"
+          :src="user.profileHeader | proxyApiUrl"
+          @error="$event.target.style.display = 'none'"
+          class="profile-header-image"
+          :alt="imageAlt"
+        />
+        <div class="profileHeader-attachments-upload-area">
+          <div class="profileHeader-drag-marker">
+            <base-icon v-if="hover" name="image" />
+          </div>
+        </div>
+      </div>
+    </vue-dropzone>
+    <div v-else>
+      <img
+        v-if="user && user.profileHeader"
+        :src="user.profileHeader | proxyApiUrl"
+        @error="$event.target.style.display = 'none'"
+        class="profile-header-image"
+        :alt="imageAlt"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import vueDropzone from 'nuxt-dropzone'
+import { updateUserMutation } from '~/graphql/User.js'
+
+export default {
+  name: 'UserProfileHeader',
+  components: {
+    vueDropzone,
+  },
+  props: {
+    user: {
+      type: Object,
+      default: null,
+    },
+    editable: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      dropzoneOptions: {
+        url: this.vddrop,
+        maxFilesize: 5.0,
+        previewTemplate: this.template(),
+      },
+      error: false,
+      hover: false,
+    }
+  },
+  computed: {
+    imageAlt() {
+      if (!this.user && !this.user.name) return 'Profile header image'
+      return 'Profile header image of ' + this.user.name
+    },
+    profileHeaderUrl() {
+      const { profileHeader } = this.user
+      return profileHeader && profileHeader.url
+    },
+  },
+  watch: {
+    error() {
+      const that = this
+      setTimeout(function () {
+        that.error = false
+      }, 2000)
+    },
+  },
+  methods: {
+    template() {
+      return `<div class="dz-preview dz-file-preview">
+                <div class="dz-image">
+                  <div data-dz-thumbnail-bg></div>
+                </div>
+              </div>
+      `
+    },
+    vddrop(file) {
+      const profileHeaderUpload = file[0]
+      this.$apollo
+        .mutate({
+          mutation: updateUserMutation(),
+          variables: {
+            profileHeader: {
+              upload: profileHeaderUpload,
+            },
+            id: this.user.id,
+          },
+        })
+        .then(() => {
+          this.$toast.success(this.$t('user.profileHeader.submitted'))
+        })
+        .catch((error) => this.$toast.error(error.message))
+    },
+    verror(file, message) {
+      if (file.status === 'error') {
+        this.error = true
+        this.$toast.error(file.status, message)
+      }
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+.profile-header {
+  height: 100%;
+  position: relative;
+  background-color: DarkGrey; /* Fallback color */
+}
+
+.profile-header-image {
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  width: 100%;
+}
+
+#profileHeaderDropzone {
+  height: 100%;
+}
+
+.dz-message {
+  height: 100%;
+}
+
+#profileHeaderDropzone .dz-preview {
+  transition: all 0.2s ease-out;
+  width: 160px;
+  display: flex;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image {
+  width: 100%;
+  height: 100%;
+  object-fit: contain;
+  overflow: hidden;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image > div {
+  width: inherit;
+  height: inherit;
+  border-radius: 50%;
+  background-size: cover;
+}
+
+#profileHeaderDropzone .dz-preview .dz-image > img {
+  width: 100%;
+}
+
+.profileHeader-attachments-upload-area {
+  height: 100%;
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+}
+
+.profileHeader-attachments-upload-button {
+  pointer-events: none;
+}
+
+.profileHeader-drag-marker {
+  position: relative;
+  width: 122px;
+  height: 122px;
+  border-radius: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  color: hsl(0, 0%, 25%);
+  transition: all 0.2s ease-out;
+  font-size: 60px;
+
+  background-color: rgba(255, 255, 255, 0.7);
```


All this is really cool dear! 🤗🍀
You haven even thought about `0.7`! 😍

Reply:
> [<img alt="DriesCruyskens" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/DriesCruyskens) **Authored by [DriesCruyskens](https://github.com/DriesCruyskens)**
_<time datetime="2020-06-23T15:36:55Z" title="Tuesday, June 23rd 2020, 5:36:55 pm +02:00">Jun 23, 2020</time>_

---


Yes I changed it because I didn't think it was visible enough depending on the chosen header image.

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-06-25T11:15:04Z" title="Thursday, June 25th 2020, 1:15:04 pm +02:00">Jun 25, 2020</time>_

---


👍🏼